internal/api/schemas/v1.yaml

```diff
@@ -21,7 +21,86 @@ paths:
           content:
             application/json:
               schema:
-                "$ref": "#/components/schemas/GatewayList"
+                "$ref": "#/components/schemas/GatewayPage"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
+    post:
+      description: 'Creates a new gateway.'
+      operationId: createGateway
+      operationId: addGateway
+      requestBody:
+        description: Gateway to add
+        required: true
+        content:
+          application/json:
+            schema:
+              "$ref": "#/components/schemas/Gateway"
+      responses:
+        '201':
+          description: gateway response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Gateway"
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/Error"
+  "/gateways/{namespace}/{name}":
```

Comment:
This endpoint structure implies that API consumers should always have a namespace for their gateway, even if that's something like `default`.

If this is the case, I'd prefer a more consistent and RESTful resource-based structure, if you're okay considering a "namespace" to be a resource/container:
- `GET /namespaces/{namespace}/gateways`
- `GET /namespaces/{namespace}/gateways/{name}`
-  `DELETE /namespaces/namespace/gateways/{name}`

Imagining we'd still need something like `GET /gateways` to retrieve all gateways that exist.

Reply:
In VM world, where we don't have K8s namespaces, would you ever need to CRUD a namespace?

Reply:
It should always map back to a pre-existing Consul namespace, that said, I like the additional way namespaces are specified in your suggestion, it feels less nasty. Also, given that in OSS Consul you don't have namespaces (other than the magical `default`), I'm thinking we can do a mapping for

`/gateways/*` that is equivalent to `/namespaces/default/gateways/*`

if we want to we could add a query parameter to that endpoint to unscope the request from a given namespace (i.e. like an `namespaces=*` parameter or something of the sort.

WDYT?

Reply:
Went ahead and did the above

Reply:
I like it!